pkg/controller/node/nodecontroller.go

```diff
@@ -496,6 +488,43 @@ func (nc *NodeController) monitorNodeStatus() error {
 	return nil
 }
 
+var nodeExistsInCloudProvider = func(nc *NodeController, nodeName string) (bool, error) {
```

Comment:
function pointer in struct is definitely better. Also define the actual function with a name. Once a test clobbers this, it'll be clobbered for any other test.